client/alloc_runner.go

```diff
@@ -323,15 +322,15 @@ func (r *AllocRunner) Run() {
 	// Start the task runners
 	r.taskLock.Lock()
 	for _, task := range tg.Tasks {
-		// Skip tasks that were restored
-		if _, ok := r.taskStatus[task.Name]; ok {
+		if _, ok := r.restored[task.Name]; ok {
```

Comment:
May be in another PR, but it might be nice to not have logic like this in Task Runner where it decides whether to run itself or not?